.github/workflows/knowledge_graph.py

```diff
@@ -185,21 +188,46 @@ def _check_md_file_contents(repo_root: Path, md_file: Path, ast: List[MdValue])
     return True
 
 
+@lru_cache()
+def _check_outbound_link(url: str):
+    # Following urls work correctly on a web browser but 404 when using python requests
+    whitelist = frozenset(["https://www.jessicahische.is/talkingtype"])
+    if url in whitelist:
+        return True
+
+    try:
+        url_status_code = requests.get(url).status_code
+    except requests.exceptions.SSLError:
+        print(f"INVALID SSL '{url}'")
+        # The url does exist but the cert is expired. I'm going to claim this is ok.
+        return True
+    if url_status_code == 404:
+        print(f"INVALID url '{url}' returned response status code '{url_status_code}'")
+        return False
+    return True
```

Comment:
if you set url_status_code to 495 (ssl cert error) at line 203 then you could drop the return at 206 and have one return at what is currently 207 that is true if the status code is a success value.

Perhaps you could hang onto the response instead of the status code and simply return [response.ok](https://requests.readthedocs.io/en/latest/api/#requests.Response.ok). You could also use ok to determine whether to print the INVALID message.

Reply:
`response.ok` will return false for all 400 codes. I'm only returning false for 404s. Around 20 urls in the knowledge base will return either 403 (requested resource is forbidden) or 406 (Not Acceptable client error) when using requests. If I inspect these urls in a web browser, they're fine.

Reply:
On second thoughts, I'll just add these 20 urls to the whitelist and drop the hammer approach.